channelserver/src/main.rs

```diff
@@ -107,24 +112,52 @@ fn channel_route(req: &HttpRequest<session::WsChannelSessionState>) -> Result<Ht
     )
 }
 
+/// Inject the FoxSec headers into all HTTP responses
+fn add_headers<'a>(resp: &'a mut HttpResponseBuilder) -> &'a mut HttpResponseBuilder {
+    resp.header(
+        "Content-Security-Policy",
+        "default-src 'none'; img-src 'self'; script-src 'self'; style-src 'self'; report-uri /__cspreport__",
+    )
+    .header("X-Content-Type-Options", "nosniff")
+    .header("X-Frame-Options", "deny")
+    .header("X-XSS-Protection", "1; mode=block")
+    .header("Strict-Transport-Security", "max-age=63072000")
+}
+
 fn heartbeat(req: &HttpRequest<session::WsChannelSessionState>) -> Result<HttpResponse, Error> {
     // if there's more to check, add it here.
     let body = json!({"status": "ok", "version": env!("CARGO_PKG_VERSION")});
-    Ok(HttpResponse::Ok()
-        .content_type("application/json")
-        .body(body.to_string()))
+    Ok(add_headers(HttpResponse::Ok().content_type("application/json")).body(body.to_string()))
 }
 
 fn lbheartbeat(req: &HttpRequest<session::WsChannelSessionState>) -> Result<HttpResponse, Error> {
     // load balance heartbeat. Doesn't matter what's returned, aside from a 200
-    Ok(HttpResponse::Ok().into())
+    Ok(add_headers(&mut HttpResponse::Ok()).finish())
 }
 
 fn show_version(req: &HttpRequest<session::WsChannelSessionState>) -> Result<HttpResponse, Error> {
     // Return the contents of the version.json file.
-    Ok(HttpResponse::Ok()
-        .content_type("application/json")
-        .body(include_str!("../version.json")))
+    Ok(
+        add_headers(&mut HttpResponse::Ok().content_type("application/json"))
+            .body(include_str!("../version.json")),
+    )
+}
+
+/// Dump the "CSP report" as a warning message.
+fn cspreport(
+    req: &HttpRequest<session::WsChannelSessionState>,
+) -> Box<Future<Item = HttpResponse, Error = Error>> {
+    use std::str;
```

Comment:
Heh, I was honestly wondering if I should put all the SecOps calls into their own file so that we could drop them in place in the future.

I think if I get to that point, though, it's probably worth just packaging them up properly.